dandi/cli/cmd_validate.py

```diff
@@ -10,30 +10,41 @@
 @devel_option(
     "--schema", help="Validate against new BIDS schema version", metavar="VERSION"
 )
-@click.option("--report", help="Specify path to write a report under.")
 @click.option(
-    "--report-flag",
+    "--report-path",
+    help="Write report under path, this option implies `--report/-r`.",
+)
+@click.option(
+    "--report",
     "-r",
     is_flag=True,
-    help="Whether to write a report under a unique path in the current directory. "
-    "Only usable if `--report` is not already used.",
+    help="Whether to write a report under a unique path in the DANDI log directory.",
 )
 @click.argument("paths", nargs=-1, type=click.Path(exists=True, dir_okay=True))
 @devel_debug_option()
 @map_to_click_exceptions
 def validate_bids(
-    paths, schema=None, devel_debug=False, report=False, report_flag=False
+    paths,
+    schema=None,
+    devel_debug=False,
+    report=False,
+    report_path="",
```

Comment:
Only `devel_debug` should have a default argument, as it's the only option that's not always present.  Giving the other arguments defaults is misleading, as those are never used.

Reply:
Apologies, but I don't understand what you mean by this. `dandi.cli.cmd_validate.validate()` also has default values for the others.

Reply:
@TheChymera When an option isn't given on the command line, click takes care of filling in the default.  This means that a value will always be passed in to the function, so the default values for `schema`, `report`, and `report_path` given in the signature will never be used.  Specifying default values for the function arguments is misleading, especially in the case of `report_path`, as click's default there is `None`, not an empty string.

`devel_debug` is an exception, as that option is only added to the command if an environment variable is present, and when the envvar isn't present, no `devel_debug` argument is passed to `validate_bids()`, so it needs a default.

I am aware that Yarik has misunderstood click's default handling in the past, leading to some command functions having superfluous argument defaults, but that's no reason to continue the practice.

Reply:
@yarikoptic better? :)